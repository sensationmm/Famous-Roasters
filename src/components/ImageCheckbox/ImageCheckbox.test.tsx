import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'

import { ImageCheckbox } from '.'

global.alert = jest.fn()

describe('ImageCheckbox component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <ImageCheckbox
        name="test"
        imageSrc="https://cdn.shopify.com/s/files/1/0632/7251/7848/products/19grams_Wild_At_Heart_copy.webp?v=1652256584"
        text="Dieser Kaffee ist lekka"
      />,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('The user can toggle the state by clicking', async () => {
    render(
      <ImageCheckbox
        name="test"
        imageSrc="https://cdn.shopify.com/s/files/1/0632/7251/7848/products/19grams_Wild_At_Heart_copy.webp?v=1652256584"
        text="Dieser Kaffee ist lekka"
        selectedText="Sehr lekka"
        toggleSelected={() => alert('works')}
      />,
    )
    const imageCheckbox = await screen.findByTestId('image-checkbox')
    expect(imageCheckbox).toBeInTheDocument()
    fireEvent.click(imageCheckbox)
    fireEvent.click(imageCheckbox)
  })
})
