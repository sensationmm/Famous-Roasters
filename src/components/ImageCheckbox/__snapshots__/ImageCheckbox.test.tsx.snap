// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCheckbox component Renders correctly 1`] = `
<div>
  <button
    class="w-64 h-80 mt-12 -top-4 relative cursor-pointer undefined"
    data-testid="image-checkbox"
  >
    <div
      class="w-64 h-64 top-0 absolute rounded-full bg-brand-grey-whisper"
    />
    <img
      alt=""
      class="absolute -top-8"
      src="https://cdn.shopify.com/s/files/1/0632/7251/7848/products/19grams_Wild_At_Heart_copy.webp?v=1652256584"
    />
    <div
      class="absolute h-32 text-center -bottom-4"
    >
      <div
        class=" font-syne font-normal text-2xl leading-8 font-semibold"
      >
        Dieser Kaffee ist lekka
      </div>
    </div>
  </button>
</div>
`;
